core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala

```diff
@@ -455,22 +455,9 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
         producer.flush()
         assertTrue("All request are complete.", responses.forall(_.isDone()))
         // Check the messages received by broker.
-        val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
-        val maxIters = 50
-        var iters = 0
-        consumer1.seek(new TopicPartition(topic, 0),0)
```

Comment:
@ijuma thanks for the reply. I rechecked the code and the seek here was required, because without that, the poll consumes only 100 records (ignores the already consumed ones), so without that seek, the assert on line 472 failed. This assert is from the original code where we used the old consumer API. I think checking 100 records on every iteration does the same thing, so I rebased that part of the code too, with that I can also remove the seek.